lib/bitvec.h

```diff
@@ -55,6 +57,48 @@ static inline int builtin_popcount(unsigned long x) { return __builtin_popcountl
 static inline int builtin_popcount(unsigned long long x) { return __builtin_popcountll(x); }
 #endif
 
+template<typename I>
+int count_trailing_zeroes(I x) {
```

Comment:
Any protection against zero?

Reply:
Added an `assert`. I did not want to add BUG_CHECK for such a low-level functionality & there are already asserts in bitvec.h.

Reply:
Thanks